src/main/java/org/opentripplanner/middleware/controllers/api/MonitoredTripController.java

```diff
@@ -36,6 +44,12 @@ MonitoredTrip preCreateHook(MonitoredTrip monitoredTrip, Request req) {
                 e
             );
         }
+        ItineraryExistenceChecker.Result checkResult = checkItineraryExistence(monitoredTrip, req);
+
+        // Replace the provided trip's itinerary with a verified, non-real-time version of it.
+        if (checkResult != null) {
+            updateTripWithVerifiedItinerary(monitoredTrip, req, checkResult.labeledResponses);
+        }
```

Comment:
According to Intellij, the preCreateHook and preUpdateHook contain duplicate code. I think this should be refactored into a single method?

Reply:
Good catch! Updated in https://github.com/ibi-group/otp-middleware/pull/77/commits/d331144c173deeb3e2e5cc8542f174d42312ff80 along with your other suggestions.